java/src/main/java/com/company/automation/fundamentals/optional_task/optional_task_01/OptionalTask01.java

```diff
@@ -0,0 +1,414 @@
+        String numbersLine;
+        String[] stringOfNumbers;
+
+        numbersLine = inputIntegersAsAString();
+
+        stringOfNumbers = numbersLine.split("\\s+");
+
+        System.out.println(Arrays.toString(stringOfNumbers));
+
+        System.out.println();
+        printTheShortestNumber(stringOfNumbers);
+        printTheLongestNumber(stringOfNumbers);
+
+        String[] arrayByIncreasingLength;
+        arrayByIncreasingLength = sortByIncreasingLength(stringOfNumbers);
+        System.out.println("\nNumbers are sorted by increasing their length:");
+        System.out.println(Arrays.toString(arrayByIncreasingLength));
+
+        String[] arrayByDecreasingLength;
+        arrayByDecreasingLength = sortByDecreasingLength(stringOfNumbers);
+        System.out.println("\nNumbers are sorted by decreasing their length:");
+        System.out.println(Arrays.toString(arrayByDecreasingLength));
+
+        printIntegersThatAreShorterThanAverageLength(stringOfNumbers);
+        printIntegersThatAreLongerThanAverageLength(stringOfNumbers);
+
+        System.out.println("\n" + Arrays.toString(stringOfNumbers));
+
+        int numberWithMinAmountOfDifferentDigits;
+        numberWithMinAmountOfDifferentDigits = getIntegerWithMinAmountOfDifferentDigits(stringOfNumbers);
+        System.out.println("\nThe first number that contains min amount of different digits: " +
+                numberWithMinAmountOfDifferentDigits + "\n");
+
+        int quantityOfNumbersWithOnlyEvenDigits;
+        quantityOfNumbersWithOnlyEvenDigits = getQuantityOfIntegersWithOnlyEvenDigits(stringOfNumbers);
+        System.out.println("Quantity of numbers which contain only even digits: " +
+                quantityOfNumbersWithOnlyEvenDigits + "\n");
+
+        int quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits;
+        quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits =
+                getQuantityOfIntegersWithEqualAmountOfEvenAndUnevenDigits(stringOfNumbers);
+        System.out.println("Quantity of numbers which contain equal amount of even and uneven digits: " +
+                quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits + "\n");
+
+        int integerWithRisingDigitsSequence;
+        integerWithRisingDigitsSequence = getIntegerWithRisingDigitsSequence(stringOfNumbers);
+        System.out.println("The first number with rising digits sequence: " + integerWithRisingDigitsSequence + "\n");
+
+        int integerWithOnlyDifferentDigits;
+        integerWithOnlyDifferentDigits = getIntegerWithOnlyDifferentDigits(stringOfNumbers);
+        System.out.println("The first number with only different digits: " + integerWithOnlyDifferentDigits + "\n");
+
+    }
+
+
+    private static String inputIntegersAsAString() {
+        boolean hasWrongCharacters;
+        boolean hasDigits;
+        String line;
+        String message;
+        Scanner in = new Scanner(System.in);
+
+        message = "Enter integer numbers using SPACE to divide them:>> ";
```

Comment:
Please mix the declaration and initialization in one line.

Reply:
Fixed